### ISSUE TITLE
Missing citeproc-php in 1.1.1

### ISSUE THREAD

**Issue by DiegoPino** [Islandora-Collaboration-Group/ISLE#271]
#### Issue description
ISLE 1.1.1 lacks citeproc-php, needed for Islandora Scholar. Because of a Bug in the citeproc submodule, when this library is not present objects that depend on this (citation, thesis) throw a FATAL PHP error and you get a White page.

#### For an issue, describe steps to reproduce the issue

System setup (OS information, software versions, etc):

Steps:
1. Upload a Thesis Object into the Citation Collection
2. See a white page

##### What's the expected result?

- An Islandora Object page.

##### What's the actual result?

- A white page and nasty FATAL errors in the apache error logs.

#### Additional details / screenshots

Temporary Solution:
```Bash
cd :/var/www/html/sites/all/libraries
ls
bookreader  colorbox  JAIL  jquery.cycle  json2  openseadragon  pdfjs  README.txt  tuque  video-js
root@b07e58dcd215:/var/www/html/sites/all/libraries# git clone https://github.com/Islandora/citeproc-php
Cloning into 'citeproc-php'...
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1263 (delta 0), reused 0 (delta 0), pack-reused 1262
Receiving objects: 100% (1263/1263), 1.34 MiB | 4.59 MiB/s, done.
Resolving deltas: 100% (39/39), done.
````

Full Solution
1.- Add the library to the deployment script somewhere here https://github.com/Islandora-Collaboration-Group/ISLE-Drupal-Build-Tools/blob/master/isle_islandora_installer.sh#L151
OR
2.- Fix the bug in retrospective (the dangers of using release versions versus 7.x-dev. 
```PHP
function citeproc_get_citeproc_php_instance($style) {
  $citeprocs = &drupal_static(__FUNCTION__, array());
  var_dump($style);
  if (!array_key_exists($style['name'], $citeprocs) && $citeproc_php_path = libraries_get_path('citeproc-php')) {
   // HERE IS THE BUG. the && makes this be skipped and then we try to return something that does not exist yet.
    require_once "$citeproc_php_path/CiteProc.php";

    global $language;

    $citeprocs[$style['name']] = new CiteProc($style['data'], $language->language);
  }

  return $citeprocs[$style['name']];
}
```
OR
3.- Upgrade to latest where the code 
https://github.com/Islandora/islandora_scholar/blob/7.x/modules/citeproc/citeproc.module#L107-L125 is not longer the same (it will fail too if the library is not there)

**Comment by bseeger** [comment]
Or maybe add it here?  Not sure:  https://github.com/Islandora-Collaboration-Group/ISLE-Drupal-Build-Tools/tree/master/isle-drush_make

Thanks for catching this, @DiegoPino!

**Comment by bseeger** [comment]
If this is a quick fix, I'd like to get this into v 1.1.2.  Thoughts @noahwsmith, @g7morris, @dwk2?

**Comment by g7morris** [comment]
@bseeger I believe this has been fixed as of the ISLE 1.3.0 release?

**Comment by g7morris** [comment]
This has been included as a fix since ISLE 1.3.0 release. Closing ticket.
